### ISSUE TITLE
False positive (?) DAR101 for argument lists with blank lines

### ISSUE THREAD

**Issue by ezwelty** [terrencepreilly/darglint#120]
For example, Sphinx (RST) requires a blank line before and after a bullet list. However, this leads to `darglint` missing all of the arguments that follow these blank lines.

Here is an example where the first argument has a bullet list, so the second argument is missed by `darglint`. The Google style docs do not mention blank lines being allowed or not, but for what it's worth, this is parsed as expected by [sphinx.ext.napoleon](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html).

`test.py`

```python
def function(x, y):
    """
    A function.

    Args:
        x (int): A very long description.

            - Either this.
            - Or that.

        y (int): Another argument.
    """
    pass
```

```bash
darglint -v 2 test.py
# test.py:function:4: DAR101: Missing parameter(s) in Docstring: - y
```

**Comment by terrencepreilly** [comment]
Sorry about not replying to this for a while, I've been rather busy.  Yes, this is a limitation in darglint.  This occurs because darglint splits by blank to determine sections -- when an argument has a blank line, it is interpreted as a new section.  See [here](https://github.com/terrencepreilly/darglint/blob/1c6ccd8351b6097fc4ae677b4b8e14544868a6fc/darglint/parse/sphinx.py#L62) for the responsible function in Sphinx, for an example.  I chose to do this because without breaking down the docstring into sections, the runtime degrades quickly.  (Before this change, some docstrings took ~30 seconds to parse.)

This can probably be solved in a couple ways.  Either by running a separate parser over the tokens to first discover sections, or by finding a better heuristic for determining sections.  I'll mark this as a bug, but I don't expect to get to it any time soon, just because it'll be a bigger thing to tackle.

**Comment by ezwelty** [comment]
In Google and Numpy style, section headers follow a strict formatting. Can these formats be represented by your existing tokens?

- google: `\n(Arguments|...)\:\n\t` -> `NEWLINE (ARGUMENTS | ... ) NEWLINE INDENT` ?
- numpy: `\n(Parameters|...)\n\-+\n`-> `NEWLINE (ARGUMENTS | ... ) NEWLINE HEADER NEWLINE` ?

From what I understand of Sphinx style, it does not have a concept of sections, but rather uses directives to tag each line with meaning.

**Comment by muxator** [comment]
Hi, I understand darglint is not actively maintained anymore, and this would not be a trivial change (thank you, btw).

I just wanted to confirm that - as of darglint **v1.8.1** - this problem is still present.
